### PR TITLE
Update Set-CsTeamsAppPermissionPolicy.md

### PR DESCRIPTION
**Pull request by xuanzhou4**
Add some usage examples and parameters description for Set-CsTeamsAppPermissionPolicy.

### CODE REVIEW

File: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md [hunk 1]
```diff
@@ -43,7 +43,65 @@ As an admin, you can use app permission policies to enable or block specific app
 ## EXAMPLES
 
 ### Example 1
-Intentionally omitted.
+
+No apps ae blocked,allow all Microsoft apps,Third-party apps,Custom apps.
```
Comment (dariomws) [inline skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md:47]:
```suggestion
This example allows all Microsoft apps, Third-party apps, and Custom apps. No apps are blocked.

```

File: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md [hunk 2]
```diff
@@ -43,7 +43,65 @@ As an admin, you can use app permission policies to enable or block specific app
...
+```powershell
+$identity = "TestTeamsAppPermissionPolicy" + (Get-Date -Format FileDateTimeUniversal)
+New-CsTeamsAppPermissionPolicy -Identity Set-$identity
+Set-CsTeamsAppPermissionPolicy -Identity Set-$identity -DefaultCatalogAppsType BlockedAppList  -DefaultCatalogApps @()-GlobalCatalogAppsType -GlobalCatalogApps @() BlockedAppList -PrivateCatalogAppsType BlockedAppList -PrivateCatalogApps @()
+```
+
+### Example 2
+
+No apps are allowed,block all Microsoft apps,Third-party apps,Custom apps.
```
Comment (dariomws) [inline skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md:56]:
```suggestion
This example blocks all Microsoft apps, Third-party apps, and Custom apps. No apps are allowed.

```

File: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md (hunk 1 above)
Comment (get-itips) [inline skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md:47]: Please move the description below the code

File: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md [hunk 3]
```diff
@@ -43,7 +43,65 @@ As an admin, you can use app permission policies to enable or block specific app
...
+```powershell
+$identity = "TestTeamsAppPermissionPolicy" + (Get-Date -Format FileDateTimeUniversal)
+New-CsTeamsAppPermissionPolicy -Identity Set-$identity
+Set-CsTeamsAppPermissionPolicy -Identity Set-$identity -DefaultCatalogAppsType AllowedAppList -DefaultCatalogApps @() -GlobalCatalogAppsType AllowedAppList -GlobalCatalogApps @() -PrivateCatalogAppsType AllowedAppList -PrivateCatalogApps @()
+```
+
+### Example 3
+
+Allow Microsoft Lists、OneNote apps that can be installed by your users and block other Microsoft apps.
```
Comment (dariomws) [inline skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md:65]:
```suggestion
This example allows Microsoft Lists and OneNote apps and blocks other Microsoft apps. Microsoft Lists and OneNote can be installed by your users.

```

File: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md [hunk 4]
```diff
@@ -43,7 +43,65 @@ As an admin, you can use app permission policies to enable or block specific app
...
+$TaskListApp = New-Object -TypeName Microsoft.Teams.Policy.Administration.Cmdlets.Core.GlobalCatalogApp -Property @{Id="57c81e84-9b7b-4783-be4e-0b7ffc0719af"}
+$OnePlanApp = New-Object -TypeName Microsoft.Teams.Policy.Administration.Cmdlets.Core.GlobalCatalogApp -Property @{Id="ca0540bf-6b61-3027-6313-a7cb4470bf1b"}
+$GlobalCatalogAppList = @($TaskListApp,$OnePlanApp)
+# set allow TaskList and OnePlan apps and block other Third-party apps
+Set-CsTeamsAppPermissionPolicy -Identity Set-$identity -GlobalCatalogAppsType AllowedAppList  -GlobalCatalogApps $GlobalCatalogAppList
+```
+
+### Example 5
+
+Allow Custom GetStartApp、TestBotApp apps that can be installed by your users and block other Custom apps.
```
Comment (dariomws) [inline skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md:94]:
```suggestion
This example allows Custom GetStartApp and TestBotApp apps and blocks other Custom apps. GetStartApp and TestBotApp can be installed by your users.

```

File: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md [hunk 5]
```diff
@@ -153,7 +211,7 @@ Accept wildcard characters: False
 ```
 
 ### -Identity
-Do not use.
+Name of App setup permission policy.If empty, all Identities will be used by default
```
Comment (dariomws) [inline skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md:214]:
```suggestion
Name of App setup permission policy. If empty, all Identities will be used by default.

```

**Review by dariomws — CHANGES_REQUESTED** [review]

File: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md (hunk 1 above)
Comment (xuanzhou4) [reply skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md]: got it,thanks！

File: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md [hunk 6]
```diff
@@ -43,7 +43,66 @@ As an admin, you can use app permission policies to enable or block specific app
...
 
 ### Example 1
-Intentionally omitted.
+
+```powershell
+$identity = "TestTeamsAppPermissionPolicy" + (Get-Date -Format FileDateTimeUniversal)
+New-CsTeamsAppPermissionPolicy -Identity Set-$identity
+Set-CsTeamsAppPermissionPolicy -Identity Set-$identity -DefaultCatalogAppsType BlockedAppList  -DefaultCatalogApps @()-GlobalCatalogAppsType -GlobalCatalogApps @() BlockedAppList -PrivateCatalogAppsType BlockedAppList -PrivateCatalogApps @()
+```
+This example allows all Microsoft apps, Third-party apps, and Custom apps. No apps are blocked.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md:52]:
```suggestion
This example allows all Microsoft apps, third-party apps, and custom apps. No apps are blocked.

```

File: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md [hunk 7]
```diff
@@ -43,7 +43,66 @@ As an admin, you can use app permission policies to enable or block specific app
...
+
+### Example 2
+
+```powershell
+$identity = "TestTeamsAppPermissionPolicy" + (Get-Date -Format FileDateTimeUniversal)
+New-CsTeamsAppPermissionPolicy -Identity Set-$identity
+Set-CsTeamsAppPermissionPolicy -Identity Set-$identity -DefaultCatalogAppsType AllowedAppList -DefaultCatalogApps @() -GlobalCatalogAppsType AllowedAppList -GlobalCatalogApps @() -PrivateCatalogAppsType AllowedAppList -PrivateCatalogApps @()
+```
+This example blocks all Microsoft apps, Third-party apps, and Custom apps. No apps are allowed.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md:61]:
```suggestion
This example blocks all Microsoft apps, third-party apps, and custom apps. No apps are allowed.

```

File: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md [hunk 8]
```diff
@@ -43,7 +43,66 @@ As an admin, you can use app permission policies to enable or block specific app
...
+$identity = "TestTeamsAppPermissionPolicy" + (Get-Date -Format FileDateTimeUniversal)
+# create a new Teams app permission policy and block all apps
+New-CsTeamsAppPermissionPolicy -Identity Set-$identity -DefaultCatalogAppsType AllowedAppList -GlobalCatalogAppsType AllowedAppList -PrivateCatalogAppsType AllowedAppList -DefaultCatalogApps @() -GlobalCatalogApps @() -PrivateCatalogApps @()
+$TaskListApp = New-Object -TypeName Microsoft.Teams.Policy.Administration.Cmdlets.Core.GlobalCatalogApp -Property @{Id="57c81e84-9b7b-4783-be4e-0b7ffc0719af"}
+$OnePlanApp = New-Object -TypeName Microsoft.Teams.Policy.Administration.Cmdlets.Core.GlobalCatalogApp -Property @{Id="ca0540bf-6b61-3027-6313-a7cb4470bf1b"}
+$GlobalCatalogAppList = @($TaskListApp,$OnePlanApp)
+# set allow TaskList and OnePlan apps and block other Third-party apps
+Set-CsTeamsAppPermissionPolicy -Identity Set-$identity -GlobalCatalogAppsType AllowedAppList  -GlobalCatalogApps $GlobalCatalogAppList
+```
+This example allows Third-party TaskList and OnePlan apps and blocks other Third-party apps.  TaskList and OnePlan can be installed by your users.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md:91]:
```suggestion
This example allows third-party TaskList and OnePlan apps and blocks other third-party apps.  TaskList and OnePlan can be installed by your users.

```

File: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md [hunk 9]
```diff
@@ -78,11 +137,11 @@ Accept wildcard characters: False
 ```
 
 ### -DefaultCatalogAppsType
-Do not use.
+Choose to allow or block the installation of Microsoft Apps.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md:140]:
```suggestion
Choose to allow or block the installation of Microsoft apps.

```

File: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md [hunk 10]
```diff
@@ -93,7 +152,7 @@ Accept wildcard characters: False
 ```
 
 ### -Description
-Do not use.
+Description of App setup permission policy.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md:155]:
```suggestion
Description of app setup permission policy.

```

File: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md [hunk 11]
```diff
@@ -123,10 +182,10 @@ Accept wildcard characters: False
 ```
 
 ### -GlobalCatalogApps
-Do not use.
+Choose which Teams apps published by a third-party that can be installed by your users.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md:185]:
```suggestion
Choose which Teams apps published by a third party can be installed by your users.

```

File: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md [hunk 12]
```diff
@@ -138,11 +197,11 @@ Accept wildcard characters: False
 ```
 
 ### -GlobalCatalogAppsType
-Do not use.
+Choose to allow or block the installation of third-party.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md:200]:
```suggestion
Choose to allow or block the installation of third-party apps.

```

**Review by JohanFreelancer9 — CHANGES_REQUESTED** [review]
Copy review - @xuanzhou4, kindly commit the suggested changes. Many thanks.

**Comment by yogkumgit** [comment]
@JohanFreelancer9 @dariomws @get-itips @tseward

**Review by JohanFreelancer9 — APPROVED** [review]

**Comment by yogkumgit** [comment]
@dariomws @get-itips @tseward

File: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md [hunk 13]
```diff
@@ -138,11 +204,12 @@ Accept wildcard characters: False
...
 
 ### -GlobalCatalogAppsType
-Do not use.
+Choose to allow or block the installation of third-party apps.
+
 
 ```yaml
 Type: String
-Parameter Sets: (All)
+Parameter Sets: AllowedAppList,BlockedAppList
```
Comment (get-itips) [inline skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md:212]: The parameters sets in the article are Identity and Instance, are these that you are mentioning here, valid parameter sets?

Comment (xuanzhou4) [reply skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md:212]: yes, you can choose one of them as GlobalCatalogAppsType

Comment (dariomws) [inline skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md:212]:
```suggestion
Parameter Sets: (All)
```

I agree with @get-itips. Parameter sets should be (All). AllowedAppList and BlockedAppList should be possible values in the description.

**Comment by yogkumgit** [comment]
@xuanzhou4 Please implement the suggested changes. Thanks

**Comment by yogkumgit** [comment]
@xuanzhou4 Is this PR still relevant, please implement he feedback else we may proceed to close it. Thanks

**Comment by xuanzhou4** [comment]
> @xuanzhou4 Is this PR still relevant, please implement he feedback else we may proceed to close it. Thanks

Hi,Which feedback are you talking about? It seems that all comments have been resolved. Thanks!

**Comment by yogkumgit** [comment]
@get-itips @dariomws

File: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md [hunk 14]
```diff
@@ -198,11 +266,11 @@ Accept wildcard characters: False
 ```
 
 ### -PrivateCatalogAppsType
-Do not use.
+Choose which custom apps can be installed by your users.
 
 ```yaml
 Type: String
-Parameter Sets: (All)
+Parameter Sets: AllowedAppList,BlockedAppList
```
Comment (dariomws) [inline skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md:273]:
```suggestion
Parameter Sets: (All)
```

File: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md [hunk 15]
```diff
@@ -78,11 +141,12 @@ Accept wildcard characters: False
...
 
 ### -DefaultCatalogAppsType
-Do not use.
+Choose to allow or block the installation of Microsoft apps.
+
 
 ```yaml
 Type: String
-Parameter Sets: (All)
+Parameter Sets: AllowedAppList,BlockedAppList
```
Comment (dariomws) [inline skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md:149]:
```suggestion
Parameter Sets: (All)
```

**Review by dariomws — CHANGES_REQUESTED** [review]
Please add the possible values (AllowedAppList, BlockedAppList) in the description of the *AppsType parameters. Thanks!

File: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md (hunk 13 above)
Comment (get-itips) [reply skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md]: One thing is a parameter set name and the other a parameter value

**Review by get-itips — CHANGES_REQUESTED** [review]
I'm ok once suggestions are applied

**Comment by xuanzhou4** [comment]
> Please add the possible values (AllowedAppList, BlockedAppList) in the description of the *AppsType parameters. Thanks!

I have applied these suggestions change and added possible values in the description. Could you please help to merge this PR? Thanks!

**Review by dariomws — APPROVED** [review]

**Comment by yogkumgit** [comment]
@tiburd Please merge this PR. Thanks

**Pull request merged**